docs/package-migration-process-guide.md

```diff
@@ -0,0 +1,201 @@
+# Package Migration Process Guide
+
+This document outlines the process for migrating a MetaMask library into the core monorepo. The migration target is assumed to comply with the requirements defined by [`metamask-module-template`](https://github.com/MetaMask/metamask-module-template) and [`module-lint`](https://github.com/MetaMask/module-lint).
+
+## Phase A: **Preparation** in the _Source Repo_
+
+### **[PR#1]** 1. Add the following migration notice to the README
+
+```markdown
+<table><tr><td><p align="center"><b>⚠️ PLEASE READ ⚠️</b></p><p align="center">This package is currently being migrated to our <a href="https://github.com/MetaMask/core"><code>core</code></a> monorepo. Please do not make any commits to this repository while this migration is taking place, as they will not be transferred over. Also, please re-open PRs that are under active development in the core repo.</p></td></tr></table>
+```
+
+- [Example PR](https://github.com/MetaMask/eth-json-rpc-provider/pull/38)
+
+### 2. Disable `dependabot` dependency upgrades
+
+### 3. Add the source repo to the ZenHub workspace repo filter so that its issues/PRs show up on the board
+
+### **[PR#2]** 4. Align dependency versions and TypeScript, ESLint, Prettier configurations with the core monorepo
+
+- If the dependency versions of the migration target are ahead of core, consider updating the core dependencies first.
+- Apply the configurations of the core monorepo to the source repo files.
+  - Preserve any TypeScript compiler flags that are enabled in the source repo but not in core.
+- Resolve all errors or issues resulting from these changes.
+- [Example PR](https://github.com/MetaMask/eth-json-rpc-provider/pull/28)
+
+### **[PR#3]** 5. Review the `metamask-module-template`, and add any missing files or elements (e.g. LICENSE)
+
+- [Example PR](https://github.com/MetaMask/eth-json-rpc-provider/pull/24)
+
+### **[PR#4]** 6. If applicable, rename the migration target package so that it is prepended by the `@metamask/` namespace
+
+- Modify the "name" field in `package.json`.
+- Update the title of the README.md.
+- Add a CHANGELOG entry for the rename.
+
+### **[PR#5]** 7. Create a new release of the migration target from the source repo
+
+- All subsequent releases of the migration target will be made from the core monorepo.
+- [Example PR](https://github.com/MetaMask/eth-json-rpc-provider/pull/29)
+
+## Phase B: **Staging** from the core monorepo's `merged-packages/` directory
+
+### **[PR#6]** 0. Migrate the source repo's git history into the `merged-packages/` temporary directory in core
```

Comment:
Why not start at 1? :)

Reply:
Fixed here 18ccc1875e0fb33b296e847c12a5882d85c355fa